core/tests/protractor_utils/ExplorationEditorTranslationTab.js

```diff
@@ -196,9 +226,30 @@ var ExplorationEditorTranslationTab = function() {
     waitFor.pageToFullyLoad();
   };
 
+  this.errorMessageElement = function() {
+    var errorMessage = element(by.css('div.error-message'));
+    return errorMessage;
+  };
+
   this.expectFeedbackTabToBeActive = function() {
     expect(element(by.css('.protractor-test-translation-feedback-tab'))[0]
     ).toEqual(element(by.css('.oppia-active-translation-tab'))[0]);
   };
+
+  this.openUploadAudioModal = function() {
+    waitFor.elementToBeClickable(
+      uploadAudioButton, 'Upload Audio button is not clickable');
+    uploadAudioButton.click();
+  };
+
+  this.saveUploadedAudio = function(wait) {
+    waitFor.elementToBeClickable(
+      this.saveUploadedAudioButton, 'Save button is not clickable');
+    this.saveUploadedAudioButton.click();
+    if (wait) {
```

Comment:
Is there an issue in waiting regardless of "wait"?

Reply:
Yes, in checking for error, we do not expect modal to close.